subsys/bluetooth/controller/ll_sw/ull_llcp_hci.c

```diff
@@ -266,70 +268,21 @@ uint8_t ll_conn_update(uint16_t handle, uint8_t cmd, uint8_t status, uint16_t in
 		return BT_HCI_ERR_UNKNOWN_CONN_ID;
 	}
 
-	if (!cmd) {
+	if (cmd == 0U) {
+		return ull_cp_conn_update(conn, interval_min, interval_max,
+					latency, timeout);
+	} else if (cmd == 2U) {
 #if defined(CONFIG_BT_CTLR_CONN_PARAM_REQ)
-		if (feature_conn_param_req(conn)) {
-			cmd++;
-		} else if (conn->lll.role) {
-			return BT_HCI_ERR_UNSUPP_REMOTE_FEATURE;
-		}
-#else /* !CONFIG_BT_CTLR_CONN_PARAM_REQ */
-		if (conn->lll.role) {
-			return BT_HCI_ERR_CMD_DISALLOWED;
-		}
-#endif /* !CONFIG_BT_CTLR_CONN_PARAM_REQ */
-	}
-
-	if (!cmd) {
-		/*
-		 * EGON TODO: fill correct info in the connection structure
-		 * as for example:
-		 *		conn->llcp_cu.win_size = 1U;
-		 *		conn->llcp_cu.win_offset_us = 0U;
-		 *		conn->llcp_cu.interval = interval_max;
-		 *		conn->llcp_cu.latency = latency;
-		 *		conn->llcp_cu.timeout = timeout;
-		 *		conn->llcp_cu.state = LLCP_CUI_STATE_USE;
-		 *		conn->llcp_cu.cmd = 1U;
-		 */
-	} else {
-#if defined(CONFIG_BT_CTLR_CONN_PARAM_REQ)
-		cmd--;
-
-		if (cmd) {
-			/*
-			 * EGON TODO: fill correct info in
-			 * the connection structure
-			 * as for example:
-			 *	conn->llcp_conn_param.status = status;
-			 *	conn->llcp_conn_param.state = cmd;
-			 *	conn->llcp_conn_param.cmd = 1U;
-			 */
+		if (status == 0U) {
+			ull_cp_conn_param_req_reply(conn);
```

Comment:
I just realized that 'status' should be passed to ull_cp_conn_param_req_reply(), here:
Re.: 5.1.7.2: "If the Host rejects this request, then the device shall shall issue an LL_REJECT_EXT_IND PDU with the ErrorCode set to a value provided by the Host. The Host shall only use the error code Unacceptable Connection Parameters (0x3B) in order to reject the request."
So, it seems to be up to the Host to provide that error code...